core/src/main/scala/org/broadinstitute/dsde/rawls/workspace/WorkspaceService.scala

```diff
@@ -2016,22 +2015,25 @@ class WorkspaceService(protected val userInfo: UserInfo, val dataSource: SlickDa
         case None =>
           val workspaceId = UUID.randomUUID.toString
           val bucketName = getBucketName(workspaceId, workspaceRequest.authorizationDomain.exists(_.nonEmpty))
-          saveNewWorkspace(workspaceId, workspaceRequest, bucketName, dataAccess).flatMap { savedWorkspace =>
-            for {
-              project <- dataAccess.rawlsBillingProjectQuery.load(RawlsBillingProjectName(workspaceRequest.namespace))
-              projectOwnerGroupEmail <- DBIO.from(samDAO.getPolicySyncStatus(SamResourceTypeNames.billingProject, project.get.projectName.value, SamBillingProjectPolicyNames.owner, userInfo).map(_.email))
-              policyEmails <- DBIO.from(samDAO.listPoliciesForResource(SamResourceTypeNames.workspace, workspaceId, userInfo).map(_.flatMap(policy =>
-                if(policy.policyName == SamWorkspacePolicyNames.projectOwner && workspaceRequest.authorizationDomain.getOrElse(Set.empty).isEmpty) {
-                  // when there isn't an auth domain, we will use the billing project admin policy email directly on workspace
-                  // resources instead of synching an extra group. This helps to keep the number of google groups a user is in below
-                  // the limit of 2000
-                  Option(WorkspaceAccessLevels.ProjectOwner -> projectOwnerGroupEmail)
-                } else {
-                  WorkspaceAccessLevels.withPolicyName(policy.policyName.value).map(_ -> policy.email)
-                }).toMap))
-              _ <- DBIO.from(gcsDAO.setupWorkspace(userInfo, project.get, policyEmails, bucketName, getLabels(workspaceRequest.authorizationDomain.getOrElse(Set.empty).toList)))
-              response <- op(SlickWorkspaceContext(savedWorkspace))
-            } yield response
+          DBIO.from(samDAO.getPolicySyncStatus(SamResourceTypeNames.billingProject, workspaceRequest.namespace, SamBillingProjectPolicyNames.owner, userInfo).map(_.email)).flatMap { projectOwnerPolicyEmail =>
+            saveNewWorkspace(workspaceId, workspaceRequest, bucketName, projectOwnerPolicyEmail, dataAccess).flatMap { savedWorkspace =>
+              for {
+                //there's potential for another perf improvement here for workspaces with auth domains. if a workspace is in an auth domain, we'll already have
```

Comment:
@MatthewBemis would you please file this as a ticket, since you have the most context? I believe all AoU workspaces have auth domains so this is relevant.

Reply:
correct ^